di-ipv-orchestrator-stub/src/main/java/uk/gov/di/ipv/stub/orc/handlers/IpvHandler.java

```diff
@@ -68,77 +69,96 @@ public class IpvHandler {
-                String vot = request.queryMap().get("votText").value();
-                String userEmailAddress = request.queryMap().get("emailAddress").value();
-                String reproveIdentityString = request.queryMap().get("reproveIdentity").value();
-                JwtBuilder.ReproveIdentityClaimValue reproveIdentityClaimValue =
-                        StringUtils.isNotBlank(reproveIdentityString)
-                                ? JwtBuilder.ReproveIdentityClaimValue.valueOf(
-                                        reproveIdentityString)
-                                : JwtBuilder.ReproveIdentityClaimValue.NOT_PRESENT;
-
-                String credentialSubject = request.queryMap().value(JSON_PAYLOAD_PARAM);
-                String evidence = request.queryMap().value(EVIDENCE_JSON_PAYLOAD_PARAM);
-                boolean duringMigration =
-                        Objects.equals(request.queryMap().value(DURING_MIGRATION), "checked");
-
-                String userId = getUserIdValue(userIdTextValue);
-
-                JWTClaimsSet claims =
-                        JwtBuilder.buildAuthorizationRequestClaims(
-                                userId,
-                                signInJourneyIdText,
-                                ORCHESTRATOR_STUB_STATE.getValue(),
-                                vtr,
-                                errorType,
-                                userEmailAddress,
-                                reproveIdentityClaimValue,
-                                environment,
-                                duringMigration,
-                                credentialSubject,
-                                evidence,
-                                vot);
-
-                SignedJWT signedJwt = JwtBuilder.createSignedJwt(claims);
-                EncryptedJWT encryptedJwt = JwtBuilder.encryptJwt(signedJwt, environment);
-                var authRequest =
-                        new AuthorizationRequest.Builder(
-                                        new ResponseType(ResponseType.Value.CODE),
-                                        new ClientID(ORCHESTRATOR_CLIENT_ID))
-                                .state(ORCHESTRATOR_STUB_STATE)
-                                .scope(new Scope("openid"))
-                                .redirectionURI(new URI(ORCHESTRATOR_REDIRECT_URL))
-                                .endpointURI(
-                                        getIpvEndpoint(environment).resolve("/oauth2/authorize"))
-                                .requestObject(EncryptedJWT.parse(encryptedJwt.serialize()))
-                                .build();
-
-                response.redirect(authRequest.toURI().toString());
+    public Route doAuthorizeError =
+            (Request request, Response response) -> {
+                var environment = request.queryMap().get(ENVIRONMENT_PARAM).value();
+                var errorType = request.queryMap().get(ERROR_TYPE_PARAM).value();
```

Comment:
As you're passing query map in to `getAuthorizeRedirect`, can you extract the error type within that method and save passing an arg?

Reply:
No - it's deliberately passed in so that in the non-error case it's ignored (notice how it's always null in the non-error case).

We could pass a boolean flag to control whether it's ignored, but it's much the same effect.